Mathlib/Order/Atoms.lean

```diff
@@ -352,14 +355,114 @@ theorem isCoatomic_iff_forall_isCoatomic_Ici [OrderTop α] :
       forall_congr' fun _ => isCoatomic_dual_iff_isAtomic.symm.trans Iff.rfl
 #align is_coatomic_iff_forall_is_coatomic_Ici isCoatomic_iff_forall_isCoatomic_Ici
 
+section StronglyAtomic
+
+variable {α : Type*} {a b : α} [Preorder α]
+
+/-- An order is strongly atomic if every nontrivial interval `[a,b]`
+contains an element covering `a`. -/
+@[mk_iff]
+class IsStronglyAtomic (α : Type*) [Preorder α] : Prop where
+  exists_covBy_le_of_lt' : ∀ (a b : α), a < b → ∃ x, a ⋖ x ∧ x ≤ b
+
+theorem LT.lt.exists_covby_le [IsStronglyAtomic α] (h : a < b) : ∃ x, a ⋖ x ∧ x ≤ b :=
+  IsStronglyAtomic.exists_covBy_le_of_lt' a b h
+
+theorem exists_covBy_le_of_lt [IsStronglyAtomic α] (h : a < b) : ∃ x, a ⋖ x ∧ x ≤ b :=
+  h.exists_covby_le
+
+/-- An order is strongly coatomic if every nontrivial interval `[a,b]`
+contains an element covered by `b`. -/
+@[mk_iff]
+class IsStronglyCoatomic (α : Type*) [Preorder α] : Prop where
+  (exists_le_covBy_of_lt' : ∀ (a b : α), a < b → ∃ x, a ≤ x ∧ x ⋖ b)
+
+theorem LT.lt.exists_le_covby [IsStronglyCoatomic α] (h : a < b) : ∃ x, a ≤ x ∧ x ⋖ b :=
+  IsStronglyCoatomic.exists_le_covBy_of_lt' a b h
+
+theorem exists_le_covBy_of_lt [IsStronglyCoatomic α] (h : a < b) : ∃ x, a ≤ x ∧ x ⋖ b :=
+  h.exists_le_covby
+
+@[simp] theorem isStronglyAtomic_dual_iff_is_stronglyCoatomic :
+    IsStronglyAtomic αᵒᵈ ↔ IsStronglyCoatomic α := by
```

Comment:
This is a bad idea as a simp lemma as it would interfere with simp's discharging of side conditions. Sorry, I'm not explaining it greatly, but I already experimented with similar simp lemmas and it was a disaster